client/blocks/payment-request/branded-buttons.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getStripeServerData } from '../stripe-utils';
+
+export const shouldUseGooglePayBrand = () => {
+	const ua = window.navigator.userAgent.toLowerCase();
+	const isChrome =
+		/chrome/.test( ua ) &&
+		! /edge|edg|opr|brave\//.test( ua ) &&
+		window.navigator.vendor === 'Google Inc.';
+	// newer versions of Brave do not have the userAgent string
+	const isBrave = isChrome && window.navigator.brave;
+	return isChrome && ! isBrave;
+};
+
+export const GooglePayButton = ( { onButtonClicked } ) => {
+	const {
+		theme = 'dark',
+		locale = 'en',
+		height = '44',
+	} = getStripeServerData().button;
+
+	const allowedTypes = [ 'short', 'long' ];
+	// Use pre-blocks settings until we merge the two distinct settings objects.
+	/* global wc_stripe_payment_request_params */
+	const { branded_type } = wc_stripe_payment_request_params.button; // eslint-disable-line camelcase
+	const type = allowedTypes.includes( branded_type ) ? branded_type : 'long'; // eslint-disable-line camelcase
+
+	// If we're using the short button type (i.e. logo only) make sure we get the logo only SVG.
+	const googlePlaySvg =
+		type === 'long'
+			? `https://www.gstatic.com/instantbuy/svg/${ theme }/${ locale }.svg`
+			: `https://www.gstatic.com/instantbuy/svg/${ theme }_gpay.svg`;
+	const [ backgroundUrl, setBackgroundUrl ] = useState( googlePlaySvg );
+
+	// Let's make sure the localized Google Pay button exists, otherwise we fall back to the
+	// english version. This test element is not used on purpose.
+	const _testImage = ( // eslint-disable-line no-unused-vars
+		<img
+			src={ backgroundUrl }
+			alt={ 'test' }
+			onError={ () => {
+				setBackgroundUrl(
+					`https://www.gstatic.com/instantbuy/svg/${ theme }/en.svg`
+				);
+			} }
+		/>
+	);
```

Comment:
Maybe we can extract this block of code to a hook, so we can use it like this:

```js
const [ backgroundUrl ] = checkImageOrDefault( googlePlaySvg, `https://www.gstatic.com/instantbuy/svg/${ theme }/en.svg` )
```

With a hook similar to this:
```js
const checkImageOrDefault = (url, defaultUrl) => {
  const [state, setState] = useState( null )

  useEffect( () => {
    const onload = () => setState( url )
    const onerror = () => setState( defaultUrl )

    var img = document.createElement( 'img' )
    img.addEventListener( 'load', onload )
    img.addEventListener( 'error', onerror )
    img.src = url

    return function cleanup() {
      img.removeEventListener( 'load', onload )
      img.removeEventListener( 'error', onerror )
      setState( null )
    }
  }, [ url, defaultUrl ] )
  
  return [ state ]
}
```

Or:
```js
const checkImageOrDefault = (url, defaultUrl) => {
  const [state, setState] = useState( url )

  useEffect( () => {
    const _img = (
      <img
        src={ url }
        onError={ () => setState( defaultUrl ) }
      />
    )
  }, [ url, defaultUrl ] )

  return [ state ]
);
```

Reply:
@diegocurbelo - since Kris is AFK, would you be able to make this change or create an issue for it?

Reply:
Made this a hook in 033c2e489a473e6297f0169b860fced274184fa1 ☺️ 

Love this suggestion btw 👏